### DISCUSSION TITLE
Parsing a markup language

### DISCUSSION THREAD

**Discussion by nik-lus** [ohmjs/ohm#292] [Help] [answered]
I'm evaluating ohm as a parser for a custom markup language. Currently I'm using nearley.js but I'm having a hard time dealing with non language specific syntax.

For example, a my typical markup language will have:

```
A bunch of arbitrary chars (EOL)
Some other chars ^^markup syntax inside special chars^^ some other chars ^^more markup syntax^^ (EOL)
other chars again(EOL)
```

Is there an example of Ohm parsing something similar? I've gone through the basic arithmetic examples but it would help me a lot if there's an example a little closer to what I'm trying to achieve. 

Any help is appreciated. Thanks,
Nik

**Comment by joshmarinacci** [comment, marked as answer]
Yeah, Ohm really needs more examples.  For of my projects I needed to pull docs out of source code.  [Here's the code](https://github.com/joshmarinacci/filament-lang/blob/master/tools/api_parser.js) that does it. I really need to write a blog on it, but the short version is I broke it up into two phases.  Phase ONE just looks for the embedded blocks based on the delimiter. `/**` and `*/` in my case. The semantics returns an array of blocks for the document.  Phase TWO then parses the structure inside of the markup.

In your case you'd need to look for the ^^ in phase one, then do a separate pass for what's inside.   I've also done markdown parsers that work similarly.

**Reply by nik-lus** [reply]
Thanks for the quick response. 

I'll play around with that example and see where it gets me. But that's pretty much it.

**Reply by pdubroy** [reply]
We'd welcome PRs to help build up the examples. I've also thought about putting together a list of patterns/recipes/howtos, and I think this would be a nice thing to include.

BTW, I think the term for this is _island grammar_:

> An island grammar consists of (1) detailed productions for the language constructs we are specifically interested
in (2) liberal productions catching all remaining constructs; and (3) a minimal set of general definitions covering
the overall structure of a program.

From [Building Documentation Generators](https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.36.3094&rep=rep1&type=pdf) by van Deursen and Kuipers
